Sources/MessagingInApp/MessagingInApp.swift

```diff
@@ -25,27 +26,58 @@ public class MessagingInApp: ModuleTopLevelObject<MessagingInAppInstance>, Messa
         Self.shared = MessagingInApp()
     }
 
+    // testing constructor
+    internal static func initialize(organizationId: String, eventListener: InAppEventListener?, implementation: MessagingInAppInstance?, sdkInitializedUtil: SdkInitializedUtil) {
+        Self.shared = MessagingInApp(implementation: implementation, sdkInitializedUtil: sdkInitializedUtil)
+
+        if let eventListener = eventListener {
+            Self.initialize(organizationId: organizationId, eventListener: eventListener)
+        } else {
+            Self.initialize(organizationId: organizationId)
+        }
+    }
+
     // Initialize SDK module
     public static func initialize(organizationId: String) {
         Self.shared.initialize(organizationId: organizationId)
     }
 
+    public static func initialize(organizationId: String, eventListener: InAppEventListener) {
+        Self.shared.initialize(organizationId: organizationId, eventListener: eventListener)
+    }
+
     public func initialize(organizationId: String) {
-        initialize() // enables features such as setting up hooks
+        guard let implementation = implementation else {
+            sdkNotInitializedAlert("CustomerIO class has not yet been initialized. Request to initialize the in-app module has been ignored.")
+            return
+        }
+
+        initialize()
+
+        implementation.initialize(organizationId: organizationId)
+    }
+
+    public func initialize(organizationId: String, eventListener: InAppEventListener) {
```

Comment:
```suggestion
   public func initialize(organizationId: String, eventListener: InAppEventListener? = nil) {
        guard let implementation = implementation else {
            sdkNotInitializedAlert("CustomerIO class has not yet been initialized. Request to initialize the in-app module has been ignored.")
            return
        }

        initialize()
        
        if eventListener == nil {
            implementation.initialize(organizationId: organizationId)
        } else {
            implementation.initialize(organizationId: organizationId, eventListener: eventListener!)
        }
    }
```

Do we need 2 methods for this? because seems like a lot of code is similar and it doesn't have `static` so doesn't seem like it is customer facing either.

Reply:
I [added some comments to the code](https://github.com/customerio/customerio-ios/pull/211/commits/da4fb58dcc6bc5526c7738f4bb61bf8048aa2bbe) to explain this. 

It's mostly because of `MessagingInAppInstance` protocol that we have the multiple initialize functions.

Reply:
Understood, thanks for clarifying that, my biggest annoyance is seeing the same exact lines of codes in both methods. 

What do you think about maybe, encapsulating 

```
guard let implementation = implementation else {
            sdkNotInitializedAlert("CustomerIO class has not yet been initialized. Request to initialize the in-app module has been ignored.")
            return
        }

        initialize()
```

in a lambda function, which returns true/false and upon its callback, trigger 

```
implementation.initialize(......)
```

On a side note, I added [a comment on another PR of yours](https://github.com/customerio/customerio-ios/pull/252#discussion_r1086535824), which holds true for this as well, regarding organization id with event listeners.